AWSTranscribeStreaming/Internal/AWSSRWebSocketAdaptor.m

```diff
@@ -0,0 +1,75 @@
+//
+// Copyright 2010-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License").
+// You may not use this file except in compliance with the License.
+// A copy of the License is located at
+//
+// http://aws.amazon.com/apache2.0
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+//
+
+#import <Foundation/Foundation.h>
+#import "AWSSRWebSocketAdaptor.h"
+#import <AWSCore/AWSCore.h>
+#import "AWSSRWebSocketDelegateAdaptor.h"
+#import "AWSTranscribeStreamingEventDecoder.h"
+#import "AWSTranscribeStreamingClientDelegate.h"
+
+@interface AWSSRWebSocketAdaptor ()
+
+@property(nonatomic, strong, readwrite) AWSSRWebSocket* webSocket;
+
+@end
+
+@implementation AWSSRWebSocketAdaptor
+
+/**
+ Initializes the adaptor with a url request
+
+ @param urlRequest  the url request you are invoking the web socket with
+ 
+ */
+- (instancetype)initWithURLRequest:(NSURLRequest *)urlRequest {
+    AWSDDLogVerbose(@"Initializing adaptor with url request %@", urlRequest);
+    if (self = [super init]) {
+        _webSocket = [[AWSSRWebSocket alloc] initWithURLRequest:urlRequest];
+
+    }
+    return self;
+}
+
+/**
+ Web socket protocol implementations
+
+ @param message the message
+*/
+- (void)send:(id)message {
+    AWSDDLogVerbose(@"Web socket %@ sends %@", _webSocket, message);
+    [_webSocket send:message];
+}
+
+- (void)connect {
+    [_webSocket open];
+}
+
+
+- (void)disconnect {
+    [_webSocket close];
+}
+
+// @optional method
```

Comment:
If what you mean by optional method? provider can optionally implement? 

If the caller does not set the delegate then it's basically saying that they do not care about whether some data is received on the socket (whether it was connected, disconnected, when some data is received on the socket). i believe this is correct behavior.

if the provider can optionally implement the "setDelegate" method, then I don't think it should be optional. the websocket provider should always implement the method so callers can call it. what happens if a websocket provider does not implement setDelegate method and caller calls setDelegate?